pkg/app/piped/apistore/eventstore/store.go

```diff
@@ -105,9 +112,10 @@ func (s *store) Run(ctx context.Context) error {
 // sync fetches a list of events newly created after its own milestone,
 // and updates the cache of latest events.
 func (s *store) sync(ctx context.Context) error {
+	// Fetch events in descending order.
 	resp, err := s.apiClient.ListEvents(ctx, &pipedservice.ListEventsRequest{
-		From:  s.milestone,
-		Order: pipedservice.ListOrder_ASC,
+		From:  time.Now().Add(-cacheTTL).Unix(),
+		Order: pipedservice.ListOrder_DESC,
```

Comment:
We only care about NOT_HANDLED events so it would be nice to have that condition to avoid storing other ones.

Reply:
Looks neat. But actually we need to care about the latest event that is likely not to be `NOT_HANDLED`.

Reply:
No matter what status events are, we have to keep the latest event on the cache.